### PR TITLE
ci: find admitted compiles beyond the first jobs page

### PR DESCRIPTION
**Pull request by teamleaderleo**
Compile-admission reuse only inspected the first 100 jobs across a prior run's attempts. When a matching successful compile fell on a later page, the candidate compiled again despite having the required same-attempt fingerprint evidence.

Read up to three job pages per candidate producer, stopping at a match or the end. API errors still fall back to compilation, and the existing repository, fingerprint and attempt checks remain in place. The lookup stays bounded when a run has many attempts.

Validation:
- The first commit adds regressions that fail on the old lookup; the second fixes them.
- Focused cases cover a page-two success, mismatched attempt fingerprints, page-two API failure, request limits, early success and an empty job list.
- Full `tests/test_ci_change_areas.py` run passed locally.

This fixes a reproduced edge-case miss; it does not claim a measured fleet-wide savings percentage. No collected historical run in the earlier sample exceeded 100 jobs.

---
<a href="https://app.blacksmith.sh/manaflow-ai/codesmith/cmux/pr/13240?ref=codesmith_pr_footer"><picture><source media="(prefers-color-scheme: dark)" srcset="https://pr-comments-assets.blacksmith.sh/codesmith/view-with-codesmith-dark-v2.svg"><source media="(prefers-color-scheme: light)" srcset="https://pr-comments-assets.blacksmith.sh/codesmith/view-with-codesmith-light-v2.svg">View with [code]smith</picture></a> <a href="https://backend.blacksmith.sh/track/enable-autofix?expires=1792506537&installation_model_id=21920&pr_number=13240&ref=codesmith_pr_footer&repository=manaflow-ai%2Fcmux&return_to=https%3A%2F%2Fgithub.com%2Fmanaflow-ai%2Fcmux%2Fpull%2F13240&signature=bed57edb91ac705e1c24b66b018143b38f3205ade88ba79f311481d1f3a44ede"><picture><source media="(prefers-color-scheme: dark)" srcset="https://pr-comments-assets.blacksmith.sh/codesmith/autofix-with-codesmith-dark.svg"><source media="(prefers-color-scheme: light)" srcset="https://pr-comments-assets.blacksmith.sh/codesmith/autofix-with-codesmith-light.svg">Autofix with [code]smith</picture></a>
<sup>Need help on this PR? Tag <code>@codesmith-bot</code> with what you need. Autofix is disabled.</sup>

---
## Summary by cubic
Fixes compile admission so it looks beyond the first page of a prior run's jobs. The old lookup only inspected the first 100 jobs per run, missing successful compiles on later pages and causing unnecessary recompiles. It now reads up to three job pages, stopping at a match or the end, and falls back to compiling when the API errors.

- Added regression tests covering a page-two success, page-two API failure, the page cap, early success, and an empty job list.
- Existing repository, fingerprint, and attempt checks are unchanged.

<sup>Written for commit 959dcea6585895f6f0a36b68c3214bc9f3299d53. Summary will update on new commits.</sup>

<a href="https://cubic.dev/pr/manaflow-ai/cmux/pull/13240?utm_source=github" target="_blank" rel="noopener noreferrer" data-no-image-dialog="true"><picture><source media="(prefers-color-scheme: dark)" srcset="https://www.cubic.dev/buttons/review-in-cubic-dark.svg"><source media="(prefers-color-scheme: light)" srcset="https://www.cubic.dev/buttons/review-in-cubic-light.svg">Review in cubic</picture></a>

## Summary by CodeRabbit

* **Bug Fixes**
  * Improved admission job lookup for workflow reruns with jobs spread across multiple result pages.
  * Added bounded pagination to check up to three pages and stop early when no additional jobs are available.
  * Preserved fallback behavior when a later page cannot be retrieved.

* **Tests**
  * Added coverage for multi-page job results, lookup failures, and the pagination limit.

### CODE REVIEW

File: tests/test_ci_change_areas.py [hunk 1]
```diff
@@ -1233,6 +1236,72 @@ def find(fingerprint: str, artifacts: list[str], jobs: list[dict]) -> str | None
...
+                raise subprocess.CalledProcessError(1, "gh")
+        return base_api(path)
+
+    assert admitted_run(api, "manaflow-ai/cmux", "feature", "abc", current_run_id=9) is None
+    assert pages == [1, 2], "the lookup must reach the failed page before falling back"
+
+
+def test_admission_lookup_bounds_job_pages_and_stops_after_a_match() -> None:
+    sys.path.insert(0, str(ROOT / "scripts/ci"))
+    from find_admitted_build import admitted_run, artifact_name
```
[the remark this answers is not in the document]
Comment (teamleaderleo) [reply tests/test_ci_change_areas.py]: Fixed in 959dcea6585895f6f0a36b68c3214bc9f3299d53. All three new tests now restore the original sys.path contents in finally immediately after importing the helper, including when import setup fails. The full tests/test_ci_change_areas.py suite passes; a focused check also verified restoration for all three tests on successful execution and injected import failure.
